cloudformation.yaml

```diff
@@ -29,14 +29,44 @@ Parameters:
     Default: 7
   SsmOutputBucketName:
     Type: String
-    Description: Bucket used to store SSM command output. This bucket must already exist and the instances which receive SSM commands must have PutObject permissions for this bucket.
+    Description: OPTIONAL! Bucket used to store SSM command output. The instances which receive SSM commands must have PutObject permissions for this bucket. If this is not provided a bucket will be created as part of this stack.
+    Default: ""
   SNSTopicForAlerts:
     Type: String
     Description: The name of the SNS topic used for alerting in the case of a failed rotation attempt.
 
 
+Conditions:
+  ShouldCreateSsmOutputBucket: !Equals
+    - !Ref SsmOutputBucketName
+    - ""
+
 Resources:
 
+  NodeRotationSsmOutputBucket:
+    Type: AWS::S3::Bucket
+    Condition: ShouldCreateSsmOutputBucket
+    Properties:
+      BucketEncryption:
```

Comment:
Could we encode the suggestion [here](https://github.com/guardian/elasticsearch-node-rotation#setting-up-node-rotations-for-an-elasticsearch-cluster) about retaining objects? I think this will be:

```
LifecycleConfiguration:
   Rules:
     - Id: DeletionRule
       Status: Enabled
       ExpirationInDays: 14
```

Reply:
done in https://github.com/guardian/elasticsearch-node-rotation/pull/73/commits/8034e67ce9772987d43ead9ed195729a11814683